File: backend/src/main/java/org/launchcode/constructionestimator/models/Item.java
package org.launchcode.constructionestimator.models;

import com.fasterxml.jackson.annotation.JsonBackReference;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.OneToMany;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Item extends NamedEntity {

    private String description;
    private int price;

    // Maybe an enum here. Will be either Fixture, appliance, or finish
    private String category;

    private String roomTypes;

    @OneToMany
    @JoinColumn(name = "item_id")
    @JsonBackReference
    private List<ItemDetails> itemDetails = new ArrayList<>();

    public Item() {}

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public List<ItemDetails> getItemDetails() {
        return itemDetails;
    }

    public void setItemDetails(List<ItemDetails> itemDetails) {
        this.itemDetails = itemDetails;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public String getRoomTypes() {
        return roomTypes;
    }

    public void setRoomTypes(String roomTypes) {
        this.roomTypes = roomTypes;
    }
}

File: frontend/src/app/item.ts

export class Item {
    id: number;
    name: string;
    room: string[];
    category: string;
    type: string;
    price: number;
    roughMaterial: number;
    labor: number;

    constructor(id: number, name: string, room: string[], category: string, type: string, price: number, roughMaterial: number, labor: number) {
        this.id = id;
        this.name = name;
        this.room = room;
        this.category = category;
        this.type = type;
        this.price = price;
        this.roughMaterial = roughMaterial;
        this.labor = labor;
    }
    
}